sympy/calculus/tests/test_util.py

```diff
@@ -63,6 +63,55 @@ def test_not_empty_in():
         Union(Interval(-2, -1, True, False), Interval(2, oo))
 
 
+def test_periodicity():
+    x = Symbol('x')
+    y = Symbol('y')
+
+    assert periodicity(sin(2*x), x) == pi
+    assert periodicity((-2)*tan(4*x), x) == pi/4
+    assert periodicity(sin(x)**2, x) == 2*pi
```

Comment:
> ```python
> In [1]: periodicity(sin(x)**y, y)
> Out[1]: 0
> ```
Should be `None`. Noted and fixed ! Thanks

> ```python
> In [2]: periodicity(sin(x)**y, k)
> Out[2]: 0
> ```

This is the correct answer since the given function is constant with respect to the variable of concern (`k`).